aws-datastore/src/test/java/com/amplifyframework/datastore/syncengine/OrchestratorTest.java

```diff
@@ -80,7 +79,7 @@ public void setup() throws AmplifyException {
 
         // SUBSCRIPTIONS_ESTABLISHED indicates that the orchestrator is up and running.
```

Comment:
The `preventConcurrentStateTransitions` test waits for `orchestratorInitObserver` before verifying that `query` is called once:

```
verify(mockApi, times(1)).query(any(), any(), any());
```

Since we are verifying that the sync query happened, it makes sense to wait for SYNC_QUERIES_READY, rather than SUBSCRIPTIONS_ESTABLISHED.  So, this change is more of a "fix".  My change to not release the semaphore sooner just uncovered the bug.

I will update the comment though!